lib/DxilContainer/DxilContainerAssembler.cpp

```diff
@@ -1788,14 +1782,23 @@ class DxilRDATWriter : public DxilPartWriter {
             payloadSizeInBytes = props.ShaderProps.Ray.paramSizeInBytes;
           }
           shaderKind = (uint32_t)props.shaderKind;
-          if (pInfo2 && DM.HasDxilEntryProps(&function)) {
+          if (DM.HasDxilEntryProps(&function)) {
             const auto &entryProps = DM.GetDxilEntryProps(&function);
-            shaderInfo = AddShaderInfo(function, entryProps, *pInfo2, flags,
-                                       TGSMInFunc[&function]);
-            if (pInfo3 && (entryProps.props.shaderKind == ShaderKind::Node ||
-                           entryProps.props.IsNode())) {
-              nodeInfo = AddShaderNodeInfo(DM, function, entryProps, *pInfo3,
+            if (pInfo2) {
+              unsigned waveSize = entryProps.props.waveSize;
+              if (waveSize) {
+                pInfo2->MinimumExpectedWaveLaneCount = waveSize;
+                pInfo2->MaximumExpectedWaveLaneCount = waveSize;
+              }
+              pInfo2->ShaderFlags = 0;
+              if (entryProps.props.IsNode()) {
+                shaderInfo = AddShaderNodeInfo(DM, function, entryProps,
+                                               *pInfo2, TGSMInFunc[&function]);
+              } else if (DXIL::CompareVersions(m_ValMajor, m_ValMinor, 1, 8) >
```

Comment:
Yes it's intentional.  AddShaderInfo adds the (now experimental) info we can do without for 1.8, since we capture the wave size and shader flags outside that function now.